packages/graph-explorer/src/connector/sparql/templates/keywordSearch/keywordSearchTemplate.test.ts

```diff
@@ -74,6 +74,32 @@ describe("SPARQL > keywordSearchTemplate", () => {
     );
   });
 
+  it("Should return a template for searched attributes without the __all predicate", () => {
+    const template = keywordSearchTemplate({
+      subjectClasses: ["air:airport"],
+      searchTerm: "JFK",
+      predicates: ["air:city", "air:code", "__all"],
+      exactMatch: true,
+    });
+
+    expect(removeExtraWhitespace(template)).toBe(
+      removeExtraWhitespace(`
+        SELECT ?subject ?pred ?value ?class { 
+          ?subject ?pred ?value { 
+            SELECT DISTINCT ?subject ?class { 
+              ?subject a ?class ; ?predicate ?value . 
+              FILTER (?predicate IN (<air:city>, <air:code>))
```

Comment:
This is a clear improvement over the inclusion of an `__all` predicate in this list, I do wonder if it makes sense to simply remove this predicate filter entirely when `all` is selected. The only potential implication I can think of is that removing this filter might lead to results which do not fit in the schema, if the schema is somehow out of date. I would however expect that removing an unneeded filter would improve the query performance somewhat.

Reply:
I'm wrestling with the same question.

In addition to the side effect you listed, the predicate list is actually filtered down from the schema to be all "searchable" predicates and with a data type of "String".

I'm not sure how those checks relate to SPARQL specifically. The filtering occurs inside the search panel logic on top of the schema object, which has already been abstracted from the underlying query engine.